.clang-format

```diff
@@ -0,0 +1,23 @@
+---
+# to see all the options in the `BasedOnStyle`, use `clang-format --dump-config`.
+# when you add a new option, make sure `comonents/example` has a valid style
+# that conforms the new option.
+
+BasedOnStyle: LLVM
+
+BreakBeforeBraces: Custom
+BraceWrapping:
+    AfterClass: false
+    AfterControlStatement: true
+    AfterEnum: true
+    AfterFunction: true
+    AfterNamespace: true
+    AfterStruct: true
+    AfterUnion: true
+    AfterExternBlock: false
+    BeforeCatch: true
+    BeforeElse: true
+    IndentBraces: false
+IndentWidth: 4
+IndentCaseLabels: true
+UseCRLF: true
```

Comment:
False. Unix-style line endings only.

Reply:
Also `SpaceBeforeParens` must be set to `ControlStatements`

Reply:
And `PointerAlignment` must be `Right`

Reply:
fixed.

as SpaceBeforeParens = SpaceBeforeParens, and PointerAlignment = Right in LLVM style, they are omitted.